buildkite/resource_test_suite.go

```diff
@@ -119,20 +183,48 @@ func (*testSuiteResource) Metadata(ctx context.Context, req resource.MetadataReq
 func (ts *testSuiteResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
 	var state testSuiteModel
 
-	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	timeout, diags := ts.client.timeouts.Read(ctx, DefaultTimeout)
+	resp.Diagnostics.Append(diags...)
+
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
-	graphqlResponse, err := getTestSuite(ctx, ts.client.genqlient, state.ID.ValueString(), 50)
+	var r *getTestSuiteResponse
+	err := retry.RetryContext(ctx, timeout, func() *retry.RetryError {
+		var err error
+		r, err = getTestSuite(ctx,
+			ts.client.genqlient, state.ID.ValueString(),
+			50,
+		)
+		if err != nil {
+			if isRetryableError(err) {
+				return retry.RetryableError(err)
+			}
+			return retry.NonRetryableError(err)
+		}
+
+		return nil
+	})
+
 	if err != nil {
-		resp.Diagnostics.AddError("Failed to load test suite from GraphQL", err.Error())
+		resp.Diagnostics.AddError(
+			"Failed to load test suite from GraphQL",
+			err.Error(),
+		)
 		return
```

Comment:
I don't think this has the remove from state logic like we've been adding everywhere. Can you add it?